testtools/UART_interface/serial_connect.py

```diff
@@ -54,7 +67,21 @@ def serial_write(ser, message, file=None):
 
     # Check that the serial connection is open
     if ser.writable():
-        bytes_written = ser.write(bytearray((message.strip()+'\r\n').encode('ascii')))
+        wait = serial_settings.mxchip_buf_pause # wait for at least 50ms between 128 byte writes.
+        buf = bytearray((message.strip()+'\r\n').encode('ascii'))
+        buf_len = len(buf) # needed for loop as buf is a destructed list
+        bytes_written = 0
+        timeout = time.time()
```

Comment:
>timeout = time.time() [](start = 8, length = 21)

This gives you the current time, not a timeout. The logic is still right it seems, but it's confusing to read. This should be `start_time = time.time()`